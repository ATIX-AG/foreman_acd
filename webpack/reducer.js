import { combineReducers } from 'redux';
import {
  cloneDeep,
  findIndex,
  findLastIndex,
} from 'lodash';

import parameterSelectionParameters from './components/ParameterSelection/ParameterSelectionReducer';
import applicationDefinitionConf from './components/ApplicationDefinition/ApplicationDefinitionReducer';
import applicationInstanceConf from './components/ApplicationInstance/ApplicationInstanceReducer';

import {
  APPLICATION_DEFINITION_PARAMETER_SELECTION_MODAL_CLOSE,
} from './components/ApplicationDefinition/ApplicationDefinitionConstants';

import {
  APPLICATION_INSTANCE_PARAMETER_SELECTION_MODAL_CLOSE,
} from './components/ApplicationInstance/ApplicationInstanceConstants';

const rootReducer = (state = {}, action: Action) => {

  let param_state = parameterSelectionParameters(state.parameterSelectionParameters, action);

  if (action.type == APPLICATION_DEFINITION_PARAMETER_SELECTION_MODAL_CLOSE) {
    action.payload.serviceParameterSelection = param_state.parameters;
  }
  let app_def_state = applicationDefinitionConf(state.applicationDefinitionConf, action);

  if (action.type == APPLICATION_INSTANCE_PARAMETER_SELECTION_MODAL_CLOSE) {
    action.payload.hostParameterSelection = param_state.parameters;
  }
  let app_ins_state = applicationInstanceConf(state.applicationInstanceConf, action);

  return {
    applicationDefinitionConf: app_def_state,
    applicationInstanceConf: app_ins_state,
    parameterSelectionParameters: param_state,
  };
};

export default rootReducer;
