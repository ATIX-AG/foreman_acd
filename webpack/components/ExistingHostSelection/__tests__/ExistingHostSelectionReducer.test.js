import { testReducerSnapshotWithFixtures } from 'react-redux-test-utils';
import reducer, { initialState } from '../ExistingHostSelectionReducer';

import {
  successState,
  initExistingHostSelectionPayload,
  existingHostSelectionSelectionChangedPayload,
  loadExistingHostSelectionSuccessPayload,
  loadExistingHostSelectionFailurePayload,
} from '../__fixtures__/existingHostSelectionReducer.fixtures';

import {
  EXISTING_HOST_SELECTION_INIT,
  EXISTING_HOST_SELECTION_LOAD_HOSTS_SUCCESS,
  EXISTING_HOST_SELECTION_LOAD_HOSTS_FAILURE,
  EXISTING_HOST_SELECTION_SELECTION_CHANGED,
} from '../ExistingHostSelectionConstants';

const fixtures = {
  'should return initial state': {
    state: initialState,
    action: {
      type: undefined,
      payload: {},
    },
  },

  'should initialize component': {
    state: initialState,
    action: {
      type: EXISTING_HOST_SELECTION_INIT,
      payload: initExistingHostSelectionPayload,
    },
  },
  'should handle changed existiong host selection': {
    state: successState,
    action: {
      type: EXISTING_HOST_SELECTION_SELECTION_CHANGED,
      payload: existingHostSelectionSelectionChangedPayload,
    },
  },
  'should load host data be successful': {
    state: successState,
    action: {
      type: EXISTING_HOST_SELECTION_LOAD_HOSTS_SUCCESS,
      payload: loadExistingHostSelectionSuccessPayload,
    },
  },
  'should load host data be erroneous': {
    state: successState,
    action: {
      type: EXISTING_HOST_SELECTION_LOAD_HOSTS_FAILURE,
      payload: loadExistingHostSelectionFailurePayload,
    },
  },
};

describe('ExistingHostSelectionReducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
