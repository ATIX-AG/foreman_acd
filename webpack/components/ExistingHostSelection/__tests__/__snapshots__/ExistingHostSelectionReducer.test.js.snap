// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExistingHostSelectionReducer should handle changed existiong host selection 1`] = `
Object {
  "allHosts": Array [
    Object {
      "ansibleParameters": Array [],
      "description": "",
      "foremanParameters": Array [],
      "hostname": "isaac-anselm",
      "id": 5,
      "service": "1",
    },
  ],
  "alreadyUsedHosts": Array [
    Object {
      "disabled": true,
      "label": "isaac-anselm",
      "tooltipText": "Host already used for this application instance",
      "value": "isaac-anselm",
    },
  ],
  "availableHosts": Array [
    Object {
      "label": "lllllll",
      "value": "lllllll",
    },
  ],
  "error": Object {
    "errorMsg": "",
    "status": "",
    "statusText": "",
  },
  "hostsInHostgroup": Object {
    "3": Object {
      "architecture_id": 1,
      "architecture_name": "x86_64",
      "build": true,
      "build_status": 2,
      "build_status_label": "Token expired",
      "capabilities": Array [
        "build",
      ],
      "certname": "isaac-anselm.mylocal",
      "comment": "",
      "compute_profile_id": null,
      "compute_profile_name": null,
      "compute_resource_id": null,
      "compute_resource_name": null,
      "created_at": "2021-08-05 11:16:34 +0200",
      "disk": "",
      "domain_id": 1,
      "domain_name": "mylocal",
      "enabled": true,
      "environment_id": null,
      "environment_name": null,
      "global_status": 0,
      "global_status_label": "Error",
      "hostgroup_id": 1,
      "hostgroup_name": "base",
      "hostgroup_title": "base",
      "id": 3,
      "image_file": "",
      "image_id": null,
      "image_name": null,
      "installed_at": null,
      "ip": null,
      "ip6": "",
      "last_compile": null,
      "last_report": null,
      "location_id": 2,
      "location_name": "Default Location",
      "mac": "00:11:22:33:44:a1",
      "managed": true,
      "medium_id": 1,
      "medium_name": "CentOS 7 mirror",
      "model_id": null,
      "model_name": null,
      "name": "isaac-anselm.mylocal",
      "operatingsystem_id": 1,
      "operatingsystem_name": "centos 7",
      "organization_id": 1,
      "organization_name": "Default Organization",
      "owner_id": 4,
      "owner_name": "Admin User",
      "owner_type": "User",
      "provision_method": "build",
      "ptable_id": 118,
      "ptable_name": "Kickstart default",
      "puppet_ca_proxy": null,
      "puppet_ca_proxy_id": null,
      "puppet_ca_proxy_name": null,
      "puppet_proxy": null,
      "puppet_proxy_id": null,
      "puppet_proxy_name": null,
      "puppet_status": 0,
      "pxe_loader": "PXELinux BIOS",
      "realm_id": null,
      "realm_name": null,
      "sp_ip": null,
      "sp_mac": null,
      "sp_name": null,
      "sp_subnet_id": null,
      "subnet6_id": null,
      "subnet6_name": null,
      "subnet_id": 1,
      "subnet_name": "mylocal",
      "updated_at": "2021-08-05 11:16:34 +0200",
      "uptime_seconds": null,
      "use_image": null,
      "uuid": null,
    },
    "7": Object {
      "architecture_id": 1,
      "architecture_name": "x86_64",
      "build": true,
      "build_status": 2,
      "build_status_label": "Token expired",
      "capabilities": Array [
        "build",
      ],
      "certname": "lllllll.mylocal",
      "comment": null,
      "compute_profile_id": null,
      "compute_profile_name": null,
      "compute_resource_id": null,
      "compute_resource_name": null,
      "created_at": "2021-08-19 14:00:20 +0200",
      "disk": null,
      "domain_id": 1,
      "domain_name": "mylocal",
      "enabled": true,
      "environment_id": null,
      "environment_name": null,
      "global_status": 0,
      "global_status_label": "Error",
      "hostgroup_id": 1,
      "hostgroup_name": "base",
      "hostgroup_title": "base",
      "id": 7,
      "image_file": "",
      "image_id": null,
      "image_name": null,
      "installed_at": null,
      "ip": null,
      "ip6": null,
      "last_compile": null,
      "last_report": null,
      "location_id": 2,
      "location_name": "Default Location",
      "mac": "00:11:22:33:60:15",
      "managed": true,
      "medium_id": 1,
      "medium_name": "CentOS 7 mirror",
      "model_id": null,
      "model_name": null,
      "name": "lllllll.mylocal",
      "operatingsystem_id": 1,
      "operatingsystem_name": "centos 7",
      "organization_id": 1,
      "organization_name": "Default Organization",
      "owner_id": 4,
      "owner_name": "Admin User",
      "owner_type": "User",
      "provision_method": "build",
      "ptable_id": 118,
      "ptable_name": "Kickstart default",
      "puppet_ca_proxy": null,
      "puppet_ca_proxy_id": null,
      "puppet_ca_proxy_name": null,
      "puppet_proxy": null,
      "puppet_proxy_id": null,
      "puppet_proxy_name": null,
      "puppet_status": 0,
      "pxe_loader": "PXELinux BIOS",
      "realm_id": null,
      "realm_name": null,
      "sp_ip": null,
      "sp_mac": null,
      "sp_name": null,
      "sp_subnet_id": null,
      "subnet6_id": null,
      "subnet6_name": null,
      "subnet_id": 1,
      "subnet_name": "mylocal",
      "updated_at": "2021-08-19 14:00:20 +0200",
      "uptime_seconds": null,
      "use_image": null,
      "uuid": null,
    },
  },
  "selectedHosts": Array [
    Object {
      "fqdn": "isaac-anselm.mylocal",
      "hostgroupId": 1,
      "hostname": "isaac-anselm",
      "serviceId": 2,
    },
  ],
  "serviceId": 2,
}
`;

exports[`ExistingHostSelectionReducer should initialize component 1`] = `
Object {
  "error": Object {
    "errorMsg": "",
    "status": "",
    "statusText": "",
  },
  "payload": Object {
    "alreadyUsedHosts": Array [
      Object {
        "disabled": true,
        "label": "isaac-anselm",
        "tooltipText": "Host already used for this application instance",
        "value": 5,
      },
    ],
    "hostsInHostgroup": Object {},
    "selectedHosts": Array [],
    "serviceId": 2,
  },
  "serviceId": undefined,
}
`;

exports[`ExistingHostSelectionReducer should load host data be erroneous 1`] = `
Object {
  "allHosts": Array [
    Object {
      "ansibleParameters": Array [],
      "description": "",
      "foremanParameters": Array [],
      "hostname": "isaac-anselm",
      "id": 5,
      "service": "1",
    },
  ],
  "alreadyUsedHosts": Array [
    Object {
      "disabled": true,
      "label": "isaac-anselm",
      "tooltipText": "Host already used for this application instance",
      "value": "isaac-anselm",
    },
  ],
  "availableHosts": Array [
    Object {
      "label": "lllllll",
      "value": "lllllll",
    },
  ],
  "error": "Something really bad happend",
  "hostsInHostgroup": Object {
    "3": Object {
      "architecture_id": 1,
      "architecture_name": "x86_64",
      "build": true,
      "build_status": 2,
      "build_status_label": "Token expired",
      "capabilities": Array [
        "build",
      ],
      "certname": "isaac-anselm.mylocal",
      "comment": "",
      "compute_profile_id": null,
      "compute_profile_name": null,
      "compute_resource_id": null,
      "compute_resource_name": null,
      "created_at": "2021-08-05 11:16:34 +0200",
      "disk": "",
      "domain_id": 1,
      "domain_name": "mylocal",
      "enabled": true,
      "environment_id": null,
      "environment_name": null,
      "global_status": 0,
      "global_status_label": "Error",
      "hostgroup_id": 1,
      "hostgroup_name": "base",
      "hostgroup_title": "base",
      "id": 3,
      "image_file": "",
      "image_id": null,
      "image_name": null,
      "installed_at": null,
      "ip": null,
      "ip6": "",
      "last_compile": null,
      "last_report": null,
      "location_id": 2,
      "location_name": "Default Location",
      "mac": "00:11:22:33:44:a1",
      "managed": true,
      "medium_id": 1,
      "medium_name": "CentOS 7 mirror",
      "model_id": null,
      "model_name": null,
      "name": "isaac-anselm.mylocal",
      "operatingsystem_id": 1,
      "operatingsystem_name": "centos 7",
      "organization_id": 1,
      "organization_name": "Default Organization",
      "owner_id": 4,
      "owner_name": "Admin User",
      "owner_type": "User",
      "provision_method": "build",
      "ptable_id": 118,
      "ptable_name": "Kickstart default",
      "puppet_ca_proxy": null,
      "puppet_ca_proxy_id": null,
      "puppet_ca_proxy_name": null,
      "puppet_proxy": null,
      "puppet_proxy_id": null,
      "puppet_proxy_name": null,
      "puppet_status": 0,
      "pxe_loader": "PXELinux BIOS",
      "realm_id": null,
      "realm_name": null,
      "sp_ip": null,
      "sp_mac": null,
      "sp_name": null,
      "sp_subnet_id": null,
      "subnet6_id": null,
      "subnet6_name": null,
      "subnet_id": 1,
      "subnet_name": "mylocal",
      "updated_at": "2021-08-05 11:16:34 +0200",
      "uptime_seconds": null,
      "use_image": null,
      "uuid": null,
    },
    "7": Object {
      "architecture_id": 1,
      "architecture_name": "x86_64",
      "build": true,
      "build_status": 2,
      "build_status_label": "Token expired",
      "capabilities": Array [
        "build",
      ],
      "certname": "lllllll.mylocal",
      "comment": null,
      "compute_profile_id": null,
      "compute_profile_name": null,
      "compute_resource_id": null,
      "compute_resource_name": null,
      "created_at": "2021-08-19 14:00:20 +0200",
      "disk": null,
      "domain_id": 1,
      "domain_name": "mylocal",
      "enabled": true,
      "environment_id": null,
      "environment_name": null,
      "global_status": 0,
      "global_status_label": "Error",
      "hostgroup_id": 1,
      "hostgroup_name": "base",
      "hostgroup_title": "base",
      "id": 7,
      "image_file": "",
      "image_id": null,
      "image_name": null,
      "installed_at": null,
      "ip": null,
      "ip6": null,
      "last_compile": null,
      "last_report": null,
      "location_id": 2,
      "location_name": "Default Location",
      "mac": "00:11:22:33:60:15",
      "managed": true,
      "medium_id": 1,
      "medium_name": "CentOS 7 mirror",
      "model_id": null,
      "model_name": null,
      "name": "lllllll.mylocal",
      "operatingsystem_id": 1,
      "operatingsystem_name": "centos 7",
      "organization_id": 1,
      "organization_name": "Default Organization",
      "owner_id": 4,
      "owner_name": "Admin User",
      "owner_type": "User",
      "provision_method": "build",
      "ptable_id": 118,
      "ptable_name": "Kickstart default",
      "puppet_ca_proxy": null,
      "puppet_ca_proxy_id": null,
      "puppet_ca_proxy_name": null,
      "puppet_proxy": null,
      "puppet_proxy_id": null,
      "puppet_proxy_name": null,
      "puppet_status": 0,
      "pxe_loader": "PXELinux BIOS",
      "realm_id": null,
      "realm_name": null,
      "sp_ip": null,
      "sp_mac": null,
      "sp_name": null,
      "sp_subnet_id": null,
      "subnet6_id": null,
      "subnet6_name": null,
      "subnet_id": 1,
      "subnet_name": "mylocal",
      "updated_at": "2021-08-19 14:00:20 +0200",
      "uptime_seconds": null,
      "use_image": null,
      "uuid": null,
    },
  },
  "selectedHosts": Array [],
  "serviceId": 2,
}
`;

exports[`ExistingHostSelectionReducer should load host data be successful 1`] = `
Object {
  "allHosts": Array [
    Object {
      "ansibleParameters": Array [],
      "description": "",
      "foremanParameters": Array [],
      "hostname": "isaac-anselm",
      "id": 5,
      "service": "1",
    },
  ],
  "alreadyUsedHosts": Array [],
  "availableHosts": Array [
    Object {
      "label": "lllllll",
      "value": "lllllll",
    },
  ],
  "error": Object {
    "errorMsg": "",
    "status": "",
    "statusText": "",
  },
  "hostsInHostgroup": Object {
    "isaac-anselm": Object {
      "architecture_id": 1,
      "architecture_name": "x86_64",
      "build": true,
      "build_status": 2,
      "build_status_label": "Token expired",
      "capabilities": Array [
        "build",
      ],
      "certname": "isaac-anselm.mylocal",
      "comment": "",
      "compute_profile_id": null,
      "compute_profile_name": null,
      "compute_resource_id": null,
      "compute_resource_name": null,
      "created_at": "2021-08-05 11:16:34 +0200",
      "disk": "",
      "domain_id": 1,
      "domain_name": "mylocal",
      "enabled": true,
      "environment_id": null,
      "environment_name": null,
      "global_status": 0,
      "global_status_label": "Error",
      "hostgroup_id": 1,
      "hostgroup_name": "base",
      "hostgroup_title": "base",
      "id": 3,
      "image_file": "",
      "image_id": null,
      "image_name": null,
      "installed_at": null,
      "ip": null,
      "ip6": "",
      "last_compile": null,
      "last_report": null,
      "location_id": 2,
      "location_name": "Default Location",
      "mac": "00:11:22:33:44:a1",
      "managed": true,
      "medium_id": 1,
      "medium_name": "CentOS 7 mirror",
      "model_id": null,
      "model_name": null,
      "name": "isaac-anselm.mylocal",
      "operatingsystem_id": 1,
      "operatingsystem_name": "centos 7",
      "organization_id": 1,
      "organization_name": "Default Organization",
      "owner_id": 4,
      "owner_name": "Admin User",
      "owner_type": "User",
      "provision_method": "build",
      "ptable_id": 118,
      "ptable_name": "Kickstart default",
      "puppet_ca_proxy": null,
      "puppet_ca_proxy_id": null,
      "puppet_ca_proxy_name": null,
      "puppet_proxy": null,
      "puppet_proxy_id": null,
      "puppet_proxy_name": null,
      "puppet_status": 0,
      "pxe_loader": "PXELinux BIOS",
      "realm_id": null,
      "realm_name": null,
      "sp_ip": null,
      "sp_mac": null,
      "sp_name": null,
      "sp_subnet_id": null,
      "subnet6_id": null,
      "subnet6_name": null,
      "subnet_id": 1,
      "subnet_name": "mylocal",
      "updated_at": "2021-08-05 11:16:34 +0200",
      "uptime_seconds": null,
      "use_image": null,
      "uuid": null,
    },
    "lllllll": Object {
      "architecture_id": 1,
      "architecture_name": "x86_64",
      "build": true,
      "build_status": 2,
      "build_status_label": "Token expired",
      "capabilities": Array [
        "build",
      ],
      "certname": "lllllll.mylocal",
      "comment": null,
      "compute_profile_id": null,
      "compute_profile_name": null,
      "compute_resource_id": null,
      "compute_resource_name": null,
      "created_at": "2021-08-19 14:00:20 +0200",
      "disk": null,
      "domain_id": 1,
      "domain_name": "mylocal",
      "enabled": true,
      "environment_id": null,
      "environment_name": null,
      "global_status": 0,
      "global_status_label": "Error",
      "hostgroup_id": 1,
      "hostgroup_name": "base",
      "hostgroup_title": "base",
      "id": 7,
      "image_file": "",
      "image_id": null,
      "image_name": null,
      "installed_at": null,
      "ip": null,
      "ip6": null,
      "last_compile": null,
      "last_report": null,
      "location_id": 2,
      "location_name": "Default Location",
      "mac": "00:11:22:33:60:15",
      "managed": true,
      "medium_id": 1,
      "medium_name": "CentOS 7 mirror",
      "model_id": null,
      "model_name": null,
      "name": "lllllll.mylocal",
      "operatingsystem_id": 1,
      "operatingsystem_name": "centos 7",
      "organization_id": 1,
      "organization_name": "Default Organization",
      "owner_id": 4,
      "owner_name": "Admin User",
      "owner_type": "User",
      "provision_method": "build",
      "ptable_id": 118,
      "ptable_name": "Kickstart default",
      "puppet_ca_proxy": null,
      "puppet_ca_proxy_id": null,
      "puppet_ca_proxy_name": null,
      "puppet_proxy": null,
      "puppet_proxy_id": null,
      "puppet_proxy_name": null,
      "puppet_status": 0,
      "pxe_loader": "PXELinux BIOS",
      "realm_id": null,
      "realm_name": null,
      "sp_ip": null,
      "sp_mac": null,
      "sp_name": null,
      "sp_subnet_id": null,
      "subnet6_id": null,
      "subnet6_name": null,
      "subnet_id": 1,
      "subnet_name": "mylocal",
      "updated_at": "2021-08-19 14:00:20 +0200",
      "uptime_seconds": null,
      "use_image": null,
      "uuid": null,
    },
  },
  "selectedHosts": Array [],
  "serviceId": 1,
}
`;

exports[`ExistingHostSelectionReducer should return initial state 1`] = `
Object {
  "error": Object {
    "errorMsg": "",
    "status": "",
    "statusText": "",
  },
  "serviceId": undefined,
}
`;
